apiserver/common/modelmanagerinterface.go

```diff
@@ -99,21 +101,13 @@ var _ ModelManagerBackend = (*modelManagerStateShim)(nil)
 
 type modelManagerStateShim struct {
 	*state.State
+	pool *state.StatePool
```

Comment:
I had them separate initially but it worked out much nicer to wrap them together. The backend interface then just represents everything to do with "state". It make the tests cleaner too.